cityofaustin/atd-moped
# Update project development status and date fields in AGOL DB view

## Associated issues
Closes https://github.com/cityofaustin/atd-data-tech/issues/18877
Closes https://github.com/cityofaustin/atd-data-tech/issues/18876
Closes https://github.com/cityofaustin/atd-data-tech/issues/18706

This PR updates the `project_development_status_date` to include the component-level phase completion dates, and it adds formats of this date that are useful for ATSD reporting.

## Testing
**URL to test:** 

Local

**Steps to test:**
1. Start up your local stack
2. Run the query below to see the new formats of the existing `project_development_status` field
   - Important note: Fiscal year starts in October so the fiscal fields use an offset on the year and quarter (i.e. calendar Quarter 4 = Fiscal Quarter 1 and calendar year October 2023 = fiscal year 2024).
3. Notice that the `project_development_status_date` value matches the component phase completion date set on this component -> https://localhost:3000/moped/projects/1566?tab=map&project_component_id=2095

```sql
SELECT
	project_id,
	project_component_id,
	project_development_status_date,
	project_development_status_date_calendar_year,
	project_development_status_date_calendar_year_month,
	project_development_status_date_calendar_year_month_numeric,
	project_development_status_date_calendar_year_quarter,
	project_development_status_date_fiscal_year,
	project_development_status_date_fiscal_year_quarter
FROM
	component_arcgis_online_view
WHERE project_id = 1566 and project_component_id = 2095;
```

---
#### Ship list
- [x] Product manager checked [DB view dependencies](https://atd-dts.gitbook.io/moped-documentation/product-management/updating-the-arcgis-online-components-database-view) - let Charlie know about the incoming update
~- [ ] Product manager added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable~
